test/unittest/transport/test_dynamic_buffer.cpp

```diff
@@ -0,0 +1,124 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#include "libcyphal/transport/dynamic_buffer.hpp"
+
+#include <gmock/gmock.h>
+
+using cetl::type_id_type;
+using cetl::rtti_helper;
+
+using DynamicBuffer = libcyphal::transport::DynamicBuffer;
+
+using testing::Return;
+using testing::StrictMock;
+
+namespace
+{
+
+class InterfaceMock : public DynamicBuffer::Interface
+{
+public:
+    MOCK_METHOD(void, moved, ());
+    MOCK_METHOD(void, deinit, ());
+
+    MOCK_METHOD(std::size_t, size, (), (const override));
+    MOCK_METHOD(std::size_t, copy, (const std::size_t, void* const, const std::size_t), (const override));
+};
+class InterfaceWrapper : public rtti_helper<type_id_type<0x01>, DynamicBuffer::Interface>
```

Comment:
Can we use randomly chosen numbers for type IDs please to avoid the risk of collisions when the test suite grows larger? If you chose 0x01 randomly then please ignore this comment.

https://xkcd.com/221

```suggestion
class InterfaceWrapper : public rtti_helper<type_id_type<0x56, 0x85>, DynamicBuffer::Interface>
```